dash/dash.py

```diff
@@ -2041,3 +2092,211 @@ def verify_url_part(served_part, url_part, part_name):
                     extra_files.append(path)
 
         self.server.run(host=host, port=port, debug=debug, **flask_run_options)
+
+    def _import_layouts_from_pages(self):
+        walk_dir = self.config.pages_folder
+        for (root, _, files) in os.walk(walk_dir):
+            for file in files:
+                if file.endswith(".py") and not file.startswith("_"):
+                    with open(os.path.join(root, file)) as f:
+                        content = f.read()
+                        if "register_page" not in content:
+                            continue
+                if file.startswith("_") or not file.endswith(".py"):
+                    continue
+                page_filename = os.path.join(root, file).replace("\\", "/")
+                _, _, page_filename = page_filename.partition("pages/")
+                page_filename = page_filename.replace(".py", "").replace("/", ".")
+                page_module = importlib.import_module(f"pages.{page_filename}")
+
+                self.page_registry = _pages.PAGE_REGISTRY.copy()
+                if f"pages.{page_filename}" in self.page_registry:
+                    self.page_registry[f"pages.{page_filename}"]["layout"] = getattr(
+                        page_module, "layout"
+                    )
+
+    def _path_to_page(self, path_id):
+        path_variables = None
+        for page in self.page_registry.values():
+            if page["path_template"]:
+                template_id = page["path_template"].strip("/")
+                path_variables = _parse_path_variables(path_id, template_id)
+                if path_variables:
+                    return page, path_variables
+            if path_id == page["path"].strip("/"):
+                return page, path_variables
+        return {}, None
+
+    def enable_pages(self):
+        if not self.pages:
+            return
+        if self.pages_folder and not os.path.exists(self.config.pages_folder):
+            warnings.warn(
+                f"A folder called {self.pages_folder} does not exist.", stacklevel=2
+            )
+        if os.path.exists(self.config.pages_folder) and self.pages:
```

Comment:
I removed the extra `self.pages` check.  I'm not sure what you mean by the other two. I'd like to suppress the warning if `pages_folder` is set to `None` because it is possible to use the `pages` features without having a pages folder.

Reply:
I didn't think of using the system without the pages directory, in that case I think it's good like that.